commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/core/LinearCombinations.java

```diff
@@ -0,0 +1,2007 @@
+                double g3 = 0.0;
+                double g2 = 0.0;
+                double g1 = 0.0;
+
+                // First traversal (from big to small).
+                double q = e3 + e2;
+                double qq = DoublePrecision.fastTwoSumLow(e3, e2, q);
+                if (qq != 0) {
+                    g3 = q;
+                    q = qq;
+                }
+                double p = q + e1;
+                qq = DoublePrecision.fastTwoSumLow(q, e1, p);
+                if (qq != 0) {
+                    g2 = p;
+                    p = qq;
+                }
+                q = p + e0;
+                qq = DoublePrecision.fastTwoSumLow(p, e0, q);
+                if (qq != 0) {
+                    g1 = p;
+                    q = qq;
+                }
+                // g0 = q
+
+                // Second traversal (from small to big)
+                // Here we do not store the round-off parts back into the expansion
+                // but summarise them into the carry using a sticky bit sum.
+                return stickySum(q, g1, g2, g3);
+            }
+
+            @Override
+            public double sum(double e0, double e1, double e2, double e3, double e4, double e5) {
+                // compress(e) into g.
+                double g5 = 0.0;
+                double g4 = 0.0;
+                double g3 = 0.0;
+                double g2 = 0.0;
+                double g1 = 0.0;
+
+                // First traversal (from big to small).
+                double q = e5 + e4;
+                double qq = DoublePrecision.fastTwoSumLow(e5, e4, q);
+                if (qq != 0) {
+                    g5 = q;
+                    q = qq;
+                }
+                double p = q + e3;
+                qq = DoublePrecision.fastTwoSumLow(q, e3, p);
+                if (qq != 0) {
+                    g4 = p;
+                    p = qq;
+                }
+                q = p + e2;
+                qq = DoublePrecision.fastTwoSumLow(p, e2, q);
+                if (qq != 0) {
+                    g3 = q;
+                    q = qq;
+                }
+                p = q + e1;
+                qq = DoublePrecision.fastTwoSumLow(q, e1, p);
+                if (qq != 0) {
+                    g2 = p;
+                    p = qq;
+                }
+                q = p + e0;
+                qq = DoublePrecision.fastTwoSumLow(p, e0, q);
+                if (qq != 0) {
+                    g1 = q;
+                    q = qq;
+                }
+                // g0 = q
+
+                // Second traversal (from small to big)
+                // Here we do not store the round-off parts back into the expansion
+                // but summarise them into the carry using a sticky bit sum.
+                return stickySum(q, g1, g2, g3, g4, g5);
+            }
+
+            @Override
+            public double sum(double e0, double e1, double e2, double e3, double e4, double e5, double e6, double e7) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 44 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1115-1144 into a separate method.